org-roam-node.el

```diff
@@ -829,23 +829,50 @@ Any top level properties drawers are incorporated into the new heading."
     (org-roam-erase-keyword "title")
     (org-roam-erase-keyword "filetags")))
 
+(defun org-roam--heading-levels ()
+  "Extract the heading level structure from the current file."
+  (org-map-entries (lambda () (org-current-level))
+                   nil 'file))
+
+(defun org-roam--can-promote-p ()
+  "Verify that this buffer is promoteable:
+There is a single level-1 heading
+and no extra content before the first heading."
+  (let ((level-1-headings-count (-count (lambda (level) (eq 1 level)) (org-roam--heading-levels)))
+        (is-top-content-empty (org-with-point-at 1
+                                (org-at-heading-p))))
+    (and
+     (eq level-1-headings-count 1)
+     is-top-content-empty)))
+
+(defun my/org-roam--outdent-headings ()
+  "Promote all the headings lower than level 1."
+  (org-map-region
+   (lambda ()
+     (when (> (org-outline-level) 1)
+       (org-do-promote)))
+   (point-min) (point-max)))
+
 (defun org-roam-promote-entire-buffer ()
   "Promote the current buffer.
-Converts a file containing a headline node at the top to a file
+
+Converts a file containing a single level-1 headline node to a file
 node."
   (interactive)
-  (org-with-point-at 1
-    (org-map-region
-     (lambda ()
-       (when (> (org-outline-level) 1)
-         (org-do-promote)))
-     (point-min) (point-max))
-    (let ((title (nth 4 (org-heading-components)))
-          (tags (nth 5 (org-heading-components))))
-      (beginning-of-line)
-      (kill-line 1)
-      (org-roam-set-keyword "title" title)
-      (when tags (org-roam-set-keyword "filetags" tags)))))
+  (save-buffer)
```

Comment:
as far as possible I want to avoid calling `save-buffer` here, is it necessary?

Reply:
I've removed it, but I'm worried that I ran into issues earlier where an unsaved buffer had node ID values in some state that broke things when the buffer was promoted. I can't reproduce the situation now, but I'll keep watching in case I can cause errors again. The `org-roam-db-update-file` is definitely needed to ensure that node state is persisted properly to the database.